internal/provider/resource_terraform_module.go

```diff
@@ -0,0 +1,318 @@
+			err.Error(),
+		)
+		return
+	}
+
+	// Copy all fields returned by Tharsis back into the plan.
+	t.copyTerraformModule(*updated, &plan)
+
+	// Set the response state to the fully-populated plan, with or without error.
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+}
+
+func (t *terraformModuleResource) Delete(ctx context.Context,
+	req resource.DeleteRequest, resp *resource.DeleteResponse) {
+
+	// Get the current state.
+	var state TerraformModuleModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Delete the Terraform module via Tharsis.
+	err := t.client.TerraformModule.DeleteModule(ctx,
+		&ttypes.DeleteTerraformModuleInput{
+			ID: state.ID.ValueString(),
+		})
+	if err != nil {
+		// Handle the case that the Terraform module no longer exists.
+		if tharsis.IsNotFoundError(err) {
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError(
+			"Error deleting Terraform module",
+			err.Error(),
+		)
+	}
+}
+
+// ImportState helps the provider implement the ResourceWithImportState interface.
+func (t *terraformModuleResource) ImportState(ctx context.Context,
+	req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+
+	// Get the Terraform module by resource path from Tharsis.
+	found, err := t.client.TerraformModule.GetModule(ctx, &ttypes.GetTerraformModuleInput{
+		Path: &req.ID,
+	})
+	if err != nil {
+		if tharsis.IsNotFoundError(err) {
+			resp.Diagnostics.AddError(
+				"Import Terraform module not found: "+req.ID,
+				"",
+			)
+			return
+		}
+		resp.Diagnostics.AddError(
+			"Import Terraform module not found: "+req.ID,
+			err.Error(),
+		)
+		return
+	}
+
+	// Import by resource path.
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("id"), found.Metadata.ID)...)
+}
+
+// copyTerraformModule copies the contents of a Terraform module.
+// It is intended to copy from a struct returned by Tharsis to a Terraform plan or state.
+func (t *terraformModuleResource) copyTerraformModule(src ttypes.TerraformModule, dest *TerraformModuleModel) {
+	dest.ID = types.StringValue(src.Metadata.ID)
+	dest.Name = types.StringValue(src.Name)
+	dest.System = types.StringValue(src.System)
+	dest.GroupPath = types.StringValue(t.resourcePathToGroupPath(src.ResourcePath))
+	dest.ResourcePath = types.StringValue(src.ResourcePath)
+	dest.RegistryNamespace = types.StringValue(src.RegistryNamespace)
+	dest.RepositoryURL = types.StringValue(src.RepositoryURL)
+	dest.Private = types.BoolValue(src.Private)
+
+	// Must use time value from SDK/API.  Using time.Now() is not reliable.
+	dest.LastUpdated = types.StringValue(src.Metadata.LastUpdatedTimestamp.Format(time.RFC850))
+}
+
+// resourcePathToGroupPath returns the group path from a resource path.
+// This function must match the GetGroupPath function in the Terraform module of the API.
+func (t *terraformModuleResource) resourcePathToGroupPath(resourcePath string) string {
```

Comment:
Most of the `resourcePathToGroupPath` functions can now be removed since the SDK now exposes the group path field.